lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1939,13 +1969,16 @@ static void checkGenericParamAccess(TypeChecker &TC,
 
 static void checkGenericParamAccess(TypeChecker &TC,
                                     const GenericParamList *params,
-                                    const ValueDecl *owner) {
-  checkGenericParamAccess(TC, params, owner, owner->getFormalAccessScope(),
+                                    const ValueDecl *owner,
+                                    bool checkUsableFromInline) {
+  checkGenericParamAccess(TC, params, owner, checkUsableFromInline,
+                          owner->getFormalAccessScope(nullptr,
+                                                      checkUsableFromInline),
                           owner->getFormalAccess());
 }
 
-/// Checks the given declaration's access to make sure it is valid given the way
-/// it is defined.
+/// Checks the given declaration's signature does not reference any other
+/// declarations that are less visible than the declaration itself.
```

Comment:
In the access control code I've tried to be good about distinguishing "is visible" from "has access to". Something declared `public` in a module that hasn't been (transitively) imported is not visible, but the client code still has access to it.

Reply:
(this one's still outstanding too)